File: vector-arithmetic/Matlab/src/mmult1.m
function D = mmult1( A, B, C )
% Matrix A times matrix B plus matrix C
% scalar level update
% GVL Algorithm 1.1.5

% Input:    A, B, C, n x n (square) matrices
% Output:   D, an n x n matrix

n = size(A);
n = n(1);

D = C;
for i = 1:n
    for j = 1:n
        for k = 1:n
            D(i,j) = D(i,j) + A(i,k) * B(k,j);
        end
    end
end
end

File: vector-arithmetic/Fortran/src/mmult1.f
        SUBROUTINE mmult (a, dima, b, dimb, c, dimc, n)
                ! Matrix A times matrix B plus matrix C
                ! scalar level update
                ! GVL algorithm 1.1.5

                ! Input:        a, b, c real n x n (square) matrices
                !               dima, dimb, dimc
                !                       integer size of each matrix
                !                       must be n or greater
                !               n       integer dimension of matrices
                ! Output:       ab + c  stored in matrix c

                IMPLICIT none
                INTEGER dima, dimb, dimc, n
                DOUBLE PRECISION a(dima, *), b(dimb, *), c(dimc, *)

                ! Internal variables
                INTEGER i, j, k

                DO i = 1,n
                        DO j = 1,n
                                ! update output matrix by adding one components
                                ! at a time to each entry
                                DO k = 1,n
                                        c(i,j) = c(i,j) + a(i,k)
     &                                          * b(k,j)
                                END DO
                        END DO
                END DO
        END SUBROUTINE
